### ISSUE TITLE
nixos-rebuild switch fails due to ipfs reinitialization

### ISSUE THREAD

**Issue by matthiasbeyer** [NixOS/nixpkgs#26069]
## Issue description

```
nixos-rebuild switch
```

produces this:

```
building Nix...
building the system configuration...
stopping the following units: nscd.service
activating the configuration...
setting up /etc...
setting up tmpfiles
reloading the following units: dbus.service
restarting the following units: polkit.service
starting the following units: nscd.service
the following new units were started: ipfs.service, systemd-vconsole-setup.service
warning: the following units failed: ipfs-init.service

● ipfs-init.service - IPFS Initializer
   Loaded: loaded (/nix/store/ic8ci6l87fw4crhkzli4z1q4ll6fnfpb-unit-ipfs-init.service/ipfs-init.service; linked; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2017-05-24 17:46:39 CEST; 27ms ago
  Process: 5866 ExecStart=/nix/store/dlxr5ghak992zqgs04as178a8k9i5c1f-unit-script/bin/ipfs-init-start (code=exited, status=1/FAILURE)
  Process: 5860 ExecStartPre=/nix/store/m9nabb1yjgcl35fl15diklmhnyln6ffs-unit-script/bin/ipfs-init-pre-start (code=exited, status=0/SUCCESS)
 Main PID: 5866 (code=exited, status=1/FAILURE)

Mai 24 17:46:39 terry systemd[1]: Starting IPFS Initializer...
Mai 24 17:46:39 terry ipfs-init-start[5866]: initializing IPFS node at /var/lib/ipfs/.ipfs
Mai 24 17:46:39 terry ipfs-init-start[5866]: Error: ipfs configuration file already exists!
Mai 24 17:46:39 terry ipfs-init-start[5866]: Reinitializing would overwrite your keys.
Mai 24 17:46:39 terry systemd[1]: ipfs-init.service: Main process exited, code=exited, status=1/FAILURE
Mai 24 17:46:39 terry systemd[1]: Failed to start IPFS Initializer.
Mai 24 17:46:39 terry systemd[1]: ipfs-init.service: Unit entered failed state.
Mai 24 17:46:39 terry systemd[1]: ipfs-init.service: Failed with result 'exit-code'.
warning: error(s) occurred while switching to the new configuration
```

I guess the ipfs initialization script does not work as intended!

### Steps to reproduce

Rebuild on unstable ... I guess simply rebuilding a working system with a slight change would trigger this error, if IPFS was enabled before and is still enabled!

## Technical details

* System: 17.09pre107265.0afb6d789c (Hummingbird)
* Nix version: nix-env (Nix) 1.11.9
* Nixpkgs version: "17.09pre107265.0afb6d789c"
* Sandboxing enabled: build-use-sandbox = true

**Comment by matthiasbeyer** [comment]
I guess / hope this will be fixed by https://github.com/NixOS/nixpkgs/pull/26064

Pinging @fpletz FYI.

**Comment by fpletz** [comment]
Hmm, maybe #25531 has something to do with that. In that case #26064 would not fix the problem.  Let me check.

**Comment by fpletz** [comment]
There is a PR to fix this issue: #25782

**Comment by matthiasbeyer** [comment]
There was another PR some time ago which tried to fix it, where I confirmed that it worked,... but now it is broken again. Let's hope that #25782 finally fixes this! :smile:

**Comment by fpletz** [comment]
Closed by #25782. The solution is to manually migrate the data (see [release notes](https://github.com/NixOS/nixpkgs/commit/df8a7d956d491eab0321319511b8ae4e808ff9c8#diff-ca92d8c04b70cab44f3b2d91f0dbd9c1R71)) or set `system.stateVersion = "17.03"`.
